app/controllers/media_objects_controller.rb

```diff
@@ -296,27 +296,34 @@ def update_status
     success_count = 0
     Array(params[:id]).each do |id|
       media_object = MediaObject.find(id)
-      if cannot? :update, media_object
-        errors += ["#{media_object.title} (#{id}) (permission denied)."]
-      else
-        case status
-          when 'publish'
-            media_object.publish!(user_key)
-            # additional save to set permalink
-            media_object.save( validate: false )
-            success_count += 1
-          when 'unpublish'
-            if can? :unpublish, media_object
-              media_object.publish!(nil)
-              success_count += 1
-            else
-              errors += ["#{media_object.title} (#{id}) (permission denied)."]
-            end
+      success = false
+      case status
+      when 'publish'
+        if can?(:update, media_object)
+          media_object.publish!(user_key)
+          # additional save to set permalink
+          media_object.save(validate: false)
+          success = true
+        end
+      when 'unpublish'
+        if can?(:unpublish, media_object)
+          media_object.publish!(nil)
+          success = true
         end
       end
+      if success
+        success_count += 1
+      else
+        errors <<
+          "#{media_object.title} (#{id}) #{t('blacklight.messages.permission_denied')}."
+      end
     end
-    message = "#{success_count} #{'media object'.pluralize(success_count)} successfully #{status}ed."
-    message += "These objects were not #{status}ed:</br> #{ errors.join('<br/> ') }" if errors.count > 0
+    message = "#{success_count} #{'media object'.pluralize(success_count)} "\
+      "successfully #{status}ed."
```

Comment:
Would i18n this like the errors are done. pass in the count/status and render the i18n. Instead of string interpolating this. This is also done in a few other places.